utils/build/docker/dotnet/parametric/Endpoints/ApmTestApi.cs

```diff
@@ -388,4 +389,21 @@ protected static string Result(object? value = null, [CallerMemberName] string?
             }
         }
     }
+
+    private static async Task AddEvent(HttpRequest request)
+    {
+        var requestJson = await ParseJsonAsync(request.Body);
+        var span = FindSpan(requestJson);
+
+        var name = requestJson.GetProperty("name").GetString() ??
+                  throw new InvalidOperationException("name is null in request json.");
+
+        var timestamp = requestJson.GetProperty("timestamp").GetInt64();
+        var attributes = requestJson.GetProperty("attributes").EnumerateObject()
+            .Select(p => new KeyValuePair<string, string>(p.Name, p.Value.GetString() ?? ""))
+            .ToList();
+
+        var spanEvent = new SpanEvent(name, DateTimeOffset.FromUnixTimeMicroseconds(timestamp), attributes);
+        span.AddEvent(spanEvent);
```

Comment:
Currently it errors as `AddEvent` is an `internal` method.

Reply:
You can access it via reflection (unsure if we are making it public?)

It'd be something like:

```csharp
Type spanType = typeof(Span);
MethodInfo addEventMethod = spanType.GetMethod("AddEvent", BindingFlags.NonPublic | BindingFlags.Instance);
addEventMethod.Invoke(span, new object[] { spanEvent });
```

Reply:
Ty Steven! I was just taking a look since I was making it public but have been deterred from doing so, I'll make the needed changes and then make sure it passes when ran against the reflection.